File: 1207.unique-number-of-occurrences/solution.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val counts = mutableMapOf<Int, Int>()
        
        arr.forEach { num -> 
            counts.put(num, (counts.get(num) ?: 0) + 1)
        }
        
        return counts.values.size == counts.values.toSet().size
    }
}

File: 1207.unique-number-of-occurrences/solution.go
func uniqueOccurrences(arr []int) bool {
	counts := make(map[int]int)
	for _, val := range arr {
			counts[val]++
	}
	
	occurrences := make(map[int]bool)
	for _, freq := range counts {
			appearred := occurrences[freq]
			if appearred {
					return false
			}
			
			occurrences[freq] = true
	}
	return true
}